lib/alertmanager_api/silence_details.py

```diff
@@ -0,0 +1,265 @@
+
+        :param instance_name: name of the server
+        :type instance_name: str
+        :param start_time_dt: start time of the silence event
+        :type start_time_dt: datetime
+        :param end_time_dt: end time of the silence event
+        :type end_time_dt: datetime
+        :param author: entity creating the silence event
+        :type author: str
+        :param comment: explanation for the silence event
+        :type comment: str
+        :param silence_id: the ID for a valid silence event
+        :type comment: str
+        :param status: the status of a valid silence event (pending, expired, active)
+        :type status: str
+        """
+        # we want to ensure start_time_dt and end_time_dt
+        # are really instances of type datetime
+        # and not just strings
+        if start_time_dt is not None and not isinstance(start_time_dt, datetime):
+            raise TypeError("start_time_dt must be of type datetime.")
+        if end_time_dt is not None and not isinstance(end_time_dt, datetime):
+            raise TypeError("end_time_dt must be of type datetime.")
+        self.instance_name = instance_name
+        self.start_time_dt = start_time_dt
+        self.end_time_dt = end_time_dt
+        self.author = author
+        self.comment = comment
+        self.silence_id = silence_id
+        self.status = status
+
+    @property
+    def is_active(self):
+        """
+        check if the current date and time is between the start time
+        and the end time of the silence event
+
+        :return: True now is between the start time and the end time
+        :rtype: bool
+        """
+        return self.status == "active"
+
+    @property
+    def is_pending(self):
+        """
+        :return: True if silence status is pending, False otherwise.
+        :rtype: bool
+        """
+        return self.status == "pending"
+
+    @property
+    def has_expired(self):
+        """
+        :return: True if silence status is expired, False otherwise.
+        :rtype: bool
+        """
+        return self.status == "expired"
+
+    @property
+    def is_valid(self):
+        """
+        :return: True if the silence has not finished yet
+        :rtype: bool
+        """
+        return self.status in ["active", "pending"]
+
+    @property
+    def start_time_str(self):
+        """
+        converts the content of self.start_time_dt to a string with format
+        "2025-01-22T11:50:00Z"
+
+        :return: the converted start date
+        :rtype: str
+        """
+        return self.start_time_dt.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    @property
+    def end_time_str(self):
+        """
+        converts the content of self.end_time_dt to a string with format
+        "2025-01-22T11:50:00Z"
+
+        :return: the converted start date
+        :rtype: str
+        """
+        return self.end_time_dt.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+class SilenceDetailsHandler(list):
```

Comment:
This is way too overengineered imo - :sweat_smile:  It's going to be too complex to maintain for such a simple purpose

I'd make a separate standalone function to parse json string from `get_silences` 
1. this will likely be the only place we'll need to ever do that - so make it a private method
2. make separate get_vaild_silences,  get_active_silences and get_hv_silences functions that will call get_silences and filter out the correct results